lib/XMLHttpRequest.js

```diff
@@ -404,7 +413,13 @@ function XMLHttpRequest(opts) {
     var ssl = false, local = false, datauri = false;
     var url;
     try {
-      url = new URL(settings.url);
+      if (settings.origin != null) {
```

Comment:
Please use `!==`, as it is more accurate

Reply:
This part is intentional as it checks for both `null` and `undefined` (as both of them indicate that user does not specify any origin)

Reply:
In that case it's probably better to use `if (!settings.origin) {`, as this will also match `false` and other falsy values.

Reply:
That works as well, will do.